dockerfiles/coda_daemon_puppeteer.py

```diff
@@ -0,0 +1,101 @@
+# This is a temporary hack for the integration test framework to be able to stop
+# and start nodes dyamically in a kubernetes environment. This script takes
+# coda arguments and will start and monitor a coda process with those arguments.
+# If a SIGUSR1 signal is sent, it will stop this process, and if a SIGUSR2 is
+# sent, it will resume the process. Since this script is a hack, there are some
+# shortcomings of the script. Most notably:
+#   - the script will stack overflow after a lot of restarts are issued
```

Comment:
if we care about fixing this, you can use a trampoline https://davywybiral.blogspot.com/2008/11/trampolining-for-recursion.html